src/components/ProviderLogin.svelte

```diff
@@ -0,0 +1,174 @@
+        controlKeysErrorMsg = '';
+        $storeProviderAccounts = [];
+        await getApi(network.endpoint?.toString() ?? '', thisDotApi, wsProvider);
+        await fetchAccounts(network, thisWeb3Enable, thisWeb3Accounts, thisDotApi.api as ApiPromise);
+        await updateConnectionStatus(thisDotApi.api as ApiPromise);
+      } catch (e) {
+        console.log(e);
+        networkErrorMsg = `Could not connect to ${
+          network.endpoint?.toString() || 'empty value'
+        }. Please enter a valid and reachable Websocket URL.`;
+        console.error(networkErrorMsg);
+      }
+      for (const account of $storeProviderAccounts) {
+        console.log(`Found provider: ${account}`);
+      }
+      if (networkErrorMsg == '' && $storeProviderAccounts.length == 0) {
+        controlKeysErrorMsg = 'No provider accounts found.  Become a provider?';
+      }
+    }
+  }
+
+  function connect() {
+    $isLoggedIn = true;
+    pageContent.dashboard();
+  }
+
+  function becomeProvider() {
+    pageContent.becomeProvider();
+  }
+
+  function networkChanged() {
+    console.log('networkChanged');
+    $storeSelectedAccount = '';
+    $transactionSigningAddress = '';
+    canConnect = false;
+  }
+
+  function accountChanged() {
+    console.log('accountChanged');
+    $transactionSigningAddress = $storeSelectedAccount;
+  }
+
+  function isValidURL(url: string): boolean {
+    try {
+      new URL(url);
+      return true;
+    } catch (_) {
+      return false;
+    }
+  }
+
+  function customNetworkChanged(event: KeyboardEvent) {
+    if (event.key === 'Enter') {
+      if (isValidURL(customNetwork)) {
+        const url = new URL(customNetwork);
+        if ($selectedNetwork != null) {
+          $selectedNetwork.endpoint = url;
+        }
+      }
+    }
+  }
+
+  function formatNetwork(network: NetworkInfo): string {
+    if (network.name === 'CUSTOM') {
+      return network.name;
+    }
+    return `${network?.name ?? ''}: ${network?.endpoint?.toString().replace(/\/$/, '') ?? ''}`;
+  }
+</script>
+
+<div id="provider-login" class="content-block flex w-single-block flex-col gap-4">
+  <BlockSection title="Provider Login">
+    <DropDownMenu
+      id="network"
+      label="Select a Network"
+      bind:selected={$selectedNetwork}
+      placeholder="Select a network"
+      options={$allNetworks}
+      onChange={networkChanged}
+      formatter={formatNetwork}
+    />
+    {#if $selectedNetwork != null && $selectedNetwork.name == 'CUSTOM'}
+      <input
+        id="other-endpoint-url"
+        type="text"
+        pattern="^(http:\/\/|https:\/\/|ws:\/\/|wss:\/\/).+"
+        placeholder="wss://some.frequency.node"
+        bind:value={customNetwork}
+        disabled={$selectedNetwork.name != 'CUSTOM'}
+        class:hidden={$selectedNetwork.name != 'CUSTOM'}
+        on:keydown={customNetworkChanged}
+      />
+    {/if}
+    <div id="network-error-msg" class="text-sm text-error">{networkErrorMsg}</div>
```

Comment:
Since this is such a large file with so much stuff going on in it, I would recommend breaking this chunk into a separate "ConnectNetwork" component.